lib/services/catalog/controllers/catalog-search-controller.js

```diff
@@ -10,8 +10,10 @@ class CatalogSearchController {
 	}
 
 	get(req, res, next) {
+		const theChannel = req.query.channel ? req.query.channel : req.identity.channel.id;
```

Comment:
Every other get always uses `req.identity.channel.id` and its not overridable. So we should follow suite here and not allow to provide a channel via `req.query`.